File: docs/1.2/core/examples/java/FiveMinuteGuide.java
import com.chain.api.*;
import com.chain.http.*;
import com.chain.signing.*;

class FiveMinuteGuide {
  public static void main(String[] args) throws Exception {
    // snippet create-client
    Client client = new Client();
    // endsnippet

    // snippet create-key
    MockHsm.Key key = MockHsm.Key.create(client);
    // endsnippet

    // snippet signer-add-key
    HsmSigner.addKey(key, MockHsm.getSignerClient(client));
    // endsnippet

    // snippet create-asset
    new Asset.Builder()
      .setAlias("gold")
      .addRootXpub(key.xpub)
      .setQuorum(1)
      .create(client);
    // endsnippet

    // snippet create-account-alice
    new Account.Builder()
      .setAlias("alice")
      .addRootXpub(key.xpub)
      .setQuorum(1)
      .create(client);
    // endsnippet

    // snippet create-account-bob
    new Account.Builder()
      .setAlias("bob")
      .addRootXpub(key.xpub)
      .setQuorum(1)
      .create(client);
    // endsnippet

    // snippet issue
    Transaction.Template issuance = new Transaction.Builder()
      .addAction(new Transaction.Action.Issue()
        .setAssetAlias("gold")
        .setAmount(100)
      ).addAction(new Transaction.Action.ControlWithAccount()
        .setAccountAlias("alice")
        .setAssetAlias("gold")
        .setAmount(100)
      ).build(client);

    Transaction.submit(client, HsmSigner.sign(issuance));
    // endsnippet

    // snippet spend
    Transaction.Template spending = new Transaction.Builder()
      .addAction(new Transaction.Action.SpendFromAccount()
        .setAccountAlias("alice")
        .setAssetAlias("gold")
        .setAmount(10))
      .addAction(new Transaction.Action.ControlWithAccount()
        .setAccountAlias("bob")
        .setAssetAlias("gold")
        .setAmount(10)
      ).build(client);

    Transaction.submit(client, HsmSigner.sign(spending));
    // endsnippet

    // snippet retire
    Transaction.Template retirement = new Transaction.Builder()
      .addAction(new Transaction.Action.SpendFromAccount()
        .setAccountAlias("bob")
        .setAssetAlias("gold")
        .setAmount(5)
      ).addAction(new Transaction.Action.Retire()
        .setAssetAlias("gold")
        .setAmount(5)
      ).build(client);

    Transaction.submit(client, HsmSigner.sign(retirement));
    // endsnippet
  }
}

File: docs/1.2/core/examples/ruby/five_minute_guide.rb
require 'chain'

# snippet create-client
chain = Chain::Client.new
# endsnippet

# snippet create-key
key = chain.mock_hsm.keys.create
# endsnippet

# snippet signer-add-key
signer = Chain::HSMSigner.new
signer.add_key(key, chain.mock_hsm.signer_conn)
# endsnippet

# snippet create-asset
chain.assets.create(
  alias: :gold,
  root_xpubs: [key.xpub],
  quorum: 1
)
# endsnippet

# snippet create-account-alice
chain.accounts.create(
  alias: :alice,
  root_xpubs: [key.xpub],
  quorum: 1
)
# endsnippet

# snippet create-account-bob
chain.accounts.create(
  alias: :bob,
  root_xpubs: [key.xpub],
  quorum: 1
)
# endsnippet

# snippet issue
issuance = chain.transactions.build do |b|
  b.issue asset_alias: :gold, amount: 100
  b.control_with_account account_alias: :alice, asset_alias: :gold, amount: 100
end

chain.transactions.submit(signer.sign(issuance))
# endsnippet

# snippet spend
spending = chain.transactions.build do |b|
  b.spend_from_account account_alias: :alice, asset_alias: :gold, amount: 10
  b.control_with_account account_alias: :bob, asset_alias: :gold, amount: 10
end

chain.transactions.submit(signer.sign(spending))
# endsnippet

# snippet retire
retirement = chain.transactions.build do |b|
  b.spend_from_account account_alias: :bob, asset_alias: :gold, amount: 5
  b.retire asset_alias: :gold, amount: 5
end

chain.transactions.submit(signer.sign(retirement))
# endsnippet
